src/router.js

```diff
@@ -1,41 +1,41 @@
-export function router(options) {
-  return function(emit) {
-    return {
-      state: {
-        router: {}
-      },
-      actions: {
-        router: {
-          set: function(state, actions, data) {
-            return {
-              router: data
-            }
+import { h } from 'hyperapp'
+
+export function router(routes) {
+  return function (app) {
+    return function (props) {
+      const actions = app(enhance(props))
+
+      window.addEventListener('popstate', function () {
+        actions.router.set({})
+      })
+
+      return actions
+
+      function enhance(props) {
+        props = Object.assign({ state: {}, actions: {} }, props)
```

Comment:
hyperapp's core aims to support IE10+, so you may need to replace `Object.assign` with a custom function to maintain that compatibility.

Maybe @JorgeBucaran would consider exporting the `merge()` function from hyperapp/src/app.js so other packages can use it and avoid duplicating efforts/code?

https://github.com/hyperapp/hyperapp/blob/master/src/app.js#L120-L129

It's also possible that his latest PR to introduce modules will mean you don't need this section of code.

Reply:
ok, i will fix that. exporting `merge` would be great. in the meantime how about simply?
```
props = props || {}
props.state = props.state || {}
props.actions = props.actions || {}
```

also, i will check out modules...